modules/data.atmosphere/R/upscale_met.R

```diff
@@ -1,87 +1,85 @@
-  # Reading in the data
-  met_data <- list()
+
+  loc.file = file.path(outfolder, paste("upscaled", basename(input_met), sep = "."))
+  if (file.exists(loc.file) && !isTRUE(overwrite)){
+    logger.severe("Output file", loc.file, "already exists. To replace it, set overwrite=TRUE")
+  }
+
+  met_lookup <- read.csv(system.file("/data/met.lookup.csv", package="PEcAn.data.atmosphere"),
+                         header = TRUE, stringsAsFactors = FALSE)
   tem <- ncdf4::nc_open(input_met)
   dim <- tem$dim
-  for (j in seq_along(var$CF.name)) {
-    if (exists(as.character(var$CF.name[j]), tem$var) == FALSE) {
-      met_data[[j]] <- NA
-    } else {
-      met_data[[j]] <- ncdf4::ncvar_get(tem, as.character(var$CF.name[j]))
+  met_data <- list()
+  met_units <- list()
+  for (v in names(tem$var)) {
+    if (!(v %in% met_lookup$CF_standard_name)) {
+      next
     }
+    met_data[[v]] <- ncdf4::ncvar_get(nc = tem, varid = v)
+    met_units[[v]] <- ncdf4::ncatt_get(nc = tem, varid = v, attname = "units")$value
   }
-  
+  met_data <- data.frame(met_data)
+
+  time_unit <- sub(" since.*", "", tem$dim$time$units)
+  time_base <- lubridate::parse_date_time(sub(".*since ", "", tem$dim$time$units),
+                                          orders = c("ymdHMSz", "ymdHMS", "ymd"))
+  time_data <- udunits2::ud.convert(tem$dim$time$vals, time_unit, reso_unit)
+
   lat_data <- as.numeric(ncdf4::ncvar_get(tem, "latitude"))
   lon_data <- as.numeric(ncdf4::ncvar_get(tem, "longitude"))
   ncdf4::nc_close(tem)
   
-  met_data <- data.frame(met_data)
-  colnames(met_data) <- var$CF.name
-  
-  if (lubridate::leap_year(year) == TRUE) {
-    sp <- 366
-  }
-  if (lubridate::leap_year(year) == FALSE) {
-    sp <- 365
-  }
-  reso_len <- sp * 24/reso
-  
-  step <- nrow(met_data)/reso_len
+  n_times <- diff(range(time_data)) / resolution
+  step <- round(nrow(met_data) / n_times, 0)
+  rows_used <- nrow(met_data) - (nrow(met_data) %% step)
+  n_steps <- (rows_used %/% step)
+  met_data <- met_data[1:rows_used,]
+  upscaled_time = colMeans(matrix(time_data[1:rows_used], nrow=step))
   upscale_data <- data.frame()
-  for (n in seq_along(var$CF.name)) {
-    upscale_data[1:reso_len,n] <- colMeans(matrix(met_data[[n]], nrow=step))
+  for (name in names(met_data)) {
+    upscale_data[1:n_steps,name] <- colMeans(matrix(met_data[[name]], nrow=step))
   }
   
-  colnames(upscale_data) <- var$CF.name
-  for (x in 1:reso_len) {
-    upscale_data$air_temperature_max[x] <- max(met_data$air_temperature[(x * step - 
-                                                                           step + 1):(x * step)])
-    upscale_data$air_temperature_min[x] <- min(met_data$air_temperature[(x * step - 
-                                                                           step + 1):(x * step)])
+  if (!is.null(upscale_data$air_temperature)
+      && is.null(upscale_data$air_temperature_max)
+      && is.null(upscale_data$air_temperature_min)) {
+    for (x in 1:n_steps) {
```

Comment:
Are there any models that use sub-daily min / max rather than mean temperature? Or can the calculation of min/ max temp be restricted to cases where upscaled resolution = 1 day?